src/base/path.cpp

```diff
@@ -52,8 +52,35 @@ const int PATHLIST_TYPEID = qRegisterMetaType<PathList>("PathList");
 
 namespace
 {
+#if defined(Q_OS_WIN)
+    bool isQStrUNCPath(const QString &path)
+    {
+        // strict format of \\host\drive\folder with no forbidden char to avoid conflict with normal path
+        if (path.left(2) != uR"(\\)"_s)
+            return false;
+        const auto lastSlash = path.lastIndexOf(u'\\');
+        if ((lastSlash - 1) <= 2)
+            return false;
+        const QRegularExpression regex {u"[\\0-\\37:?\"*<>|:/]"_s}; // no drive letter allowed C:/
+        return !regex.match(path.data()).hasMatch();
+    }
+#endif
+
     QString cleanPath(const QString &path)
     {
+#if defined(Q_OS_WIN)
+        // QT's cleanPath does not support UNC path
```

Comment:
Another idea.
You could try to split the UNC path into 2 fields. First is `host-name` and the second is the rest of the path. `host-name` will have its own QString variable in this class and the second part will be stored in `m_pathStr`. The second part will look like a relative path.
This way you could reuse existing code and you only need to concat the `host-name` back when full path is expected in various methods.

https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/62e862f4-2a51-452e-8eeb-dc4ff5ee33cc